Lib/test/test_re.py

```diff
@@ -1470,11 +1470,11 @@ def test_ascii_and_unicode_flag(self):
             self.assertIsNone(pat.match(b'\xe0'))
         # Incompatibilities
         self.assertRaises(ValueError, re.compile, br'\w', re.UNICODE)
-        self.assertRaises(ValueError, re.compile, br'(?u)\w')
+        self.assertRaises(re.error, re.compile, br'(?u)\w')
```

Comment:
Why the change in exception raised?  Isn't that an API break?

Reply:
I consider this as minor implementation detail. re.error looks more correct to me here. ValueError is related to flags specified as a separate argument, re.error is related to inline flags.

Reply:
I'm concerned that this may break some code.  For example, in Mailman we sometimes have regular expression strings provided by a site admin or list owner.  We have to check to see if they are valid by trying to compile them and catching any exceptions.  So it seems like if this changes, those checks would fail.

Reply:
`re.error` is an exception raised when an error in a regular expression is found. If you try to compile arbitrary string regular expression you should catch `re.error`. You should also catch `ValueError` (only raised when specify incompatible flags as a separate argument) and `OverflowError` (only raised if specify too large integers in `{m,n}`). And I think this is all. Now new exception added. If you have different handlers for `re.error` and `ValueError`, this change can be visible to you.